ykcapi/yk_internal.h

```diff
@@ -0,0 +1,5 @@
+// C ABI functions used internally by the system to call from C++ back to Rust.
+// They may also be used by C tests.
+
+size_t __yktrace_irtrace_len(void *trace);
+void __yktrace_irtrace_get(void *trace, size_t idx, char **res_func, size_t *res_bb);
```

Comment:
You need to wrap this in an `extern "C"` block to avoid C++ mangling the names.

Reply:
damn, I didn't realise that you had to do it on the headers, but of course!